code/modules/mob/living/simple_animal/abnormality/aleph/distortedform.dm

```diff
@@ -0,0 +1,1721 @@
+	icon = initial(icon)
+	icon_state = initial(icon_state)
+	pixel_x = initial(pixel_x)
+	base_pixel_x = initial(base_pixel_x)
+	pixel_y = initial(pixel_y)
+	base_pixel_y = initial(base_pixel_y)
+	UpdateConsoleDisplay(initial(name), initial(threat_level))
+	if(connected_door) //doors can be destroyed
+		connected_door.name = "Distorted Form containment zone" //Initial() turns it into "vault door"
+	connected_panel.name = "Distorted Form's containment panel"
+	transformed = FALSE
+	transform_timer = addtimer(CALLBACK(src, .proc/Transform), 3 MINUTES, TIMER_STOPPABLE & TIMER_OVERRIDE)
+	update_icon()
+
+/mob/living/simple_animal/hostile/abnormality/distortedform/proc/AbnoListGen() // List of possible transformations
+	. = list()
+	if(LAZYLEN(SSlobotomy_corp.all_abnormality_datums))
+		for(var/datum/abnormality/A in SSlobotomy_corp.all_abnormality_datums)
+			if(isnull(A.current))
+				continue
+			if(A.abno_path in transform_blacklist)
+				continue
+			if(A.name != name) //lets not turn into ourselves
+				. += A
+	return
+
+//Qlipthoth Stuff
+/mob/living/simple_animal/hostile/abnormality/distortedform/proc/OnMeltdownFinish(datum/source, datum/abnormality/abno_datum, worked)
+	SIGNAL_HANDLER
+	if(!worked)
+		datum_reference.qliphoth_change(-1)
+
+/mob/living/simple_animal/hostile/abnormality/distortedform/proc/CauseMelts(datum/source, datum/abnormality/abno_datum, worked)
+	var/meltdown_text = "Horrible screeches have caused a disturbance in the containment zones of the following abnormalities:"
+	var/meltdown_sound = pick("sound/abnormalities/distortedform/screech3.ogg","sound/abnormalities/distortedform/screech4.ogg")
+	var/player_count = 0
+	for(var/mob/player in GLOB.player_list)
+		if(isliving(player) && (player.mind?.assigned_role in GLOB.security_positions))
+			player_count += 1.5
+	player_count = round(player_count) + (player_count > round(player_count) ? TRUE : FALSE) // Trying to round up
+	meltdowns = SSlobotomy_corp.InitiateMeltdown(clamp(rand(player_count*0.5, player_count), 1, 10), TRUE, MELTDOWN_BLACK, 45, 60, meltdown_text, meltdown_sound)
+	for(var/obj/machinery/computer/abnormality/A in meltdowns) // TODO: Figure out a way to exclude it entirely from melts so it doesn't breach itself.
+		if(A == datum_reference.console) // TODO: This shouldn't be neccessary when I fix the above comment
+			continue
+		RegisterSignal(A, COMSIG_MELTDOWN_FINISHED, .proc/SpecialMeltFinish)
+
+/mob/living/simple_animal/hostile/abnormality/distortedform/proc/SpecialMeltFinish(datum/source, datum/abnormality/abno_datum, worked)
+	SIGNAL_HANDLER
+	if(!LAZYLEN(meltdowns))
+		return
+	meltdowns -= source
+	UnregisterSignal(source, COMSIG_MELTDOWN_FINISHED)
+	var/list/breaching_list = shuffle(GLOB.abnormality_mob_list)
+	if(!worked)
+		for(var/mob/living/simple_animal/hostile/abnormality/A in breaching_list)
+			if(A.datum_reference.qliphoth_meter > 0 && A.IsContained() && A.z == z)
+				A.datum_reference.qliphoth_change(-200)
+				return
+			return
```

Comment:
Grabs the first one int he list, yeah? Won't this never go to the second+ mob in the breaching_list due to it returning after each attempt? This lets it attempt to breach a Zayin, for example, and because it meets none of this criteria it does nothing.

Reply:
Good find!